File: src/main/java/me/muksc/tacztweaks/mixin/accessor/CodecConfigAccessor.java
package me.muksc.tacztweaks.mixin.accessor;

import dev.isxander.yacl3.config.v3.CodecConfig;
import dev.isxander.yacl3.config.v3.ReadonlyConfigEntry;
import org.spongepowered.asm.mixin.Mixin;
import org.spongepowered.asm.mixin.gen.Accessor;

import java.util.List;

@Mixin(value = CodecConfig.class, remap = false)
public interface CodecConfigAccessor {
    @Accessor
    List<ReadonlyConfigEntry<?>> getEntries();
}

File: src/main/kotlin/me/muksc/tacztweaks/mixininterop/CodecConfigAccessor.kt
package me.muksc.tacztweaks.mixininterop

import dev.isxander.yacl3.config.v3.ReadonlyConfigEntry
import me.muksc.tacztweaks.mixin.accessor.CodecConfigAccessor

inline val CodecConfigAccessor.entries: List<ReadonlyConfigEntry<*>>
    get() = `tacztweaks$getEntries`()